src/main/java/org/mockito/internal/MockitoCore.java

```diff
@@ -82,7 +96,8 @@ public IOngoingStubbing stub() {
         } else if (!mockUtil.isMock(mock)) {
             reporter.notAMockPassedToVerify(mock.getClass());
         }
-        mockingProgress.verificationStarted(new MockAwareVerificationMode(mock, mode));
+        VerificationMode actualMode = this.verificationState.verifyLazily(mode);
+        mockingProgress.verificationStarted(new MockAwareVerificationMode(mock, actualMode));
```

Comment:
How about we move the setting of a thread local state to the JUnit rule for consistency? It's kind of awkward that getter method 'isTypeMockable' actually updates the thread local state.